### ISSUE TITLE
Application is Crashing when Questionnaire.Item.TypeCode is qual to GROUP

### ISSUE THREAD

**Issue by MuhammadSalman-7214** [google/android-fhir#271]
In MoreQuestionaireItemTypes Class that uses the function
 **fun Questionnaire.Item.EnableWhen.AnswerX.getValueForType(
    type: Questionnaire.Item.TypeCode
): Message**
when the passed input to function is **GROUP**, then the application crashes because GROUP case hasn't been handled yet and it throws exception as "Unsupported value type GROUP"

**Steps to Reproduce:** 
Run the dataCapturegallery application, tap on the COVID-19 Case Report option, and the application will crash

**Stack trace of crash**
```
FATAL EXCEPTION: main
    Process: com.google.android.fhir.datacapture.gallery, PID: 20022
    java.lang.IllegalArgumentException: Unsupported value type GROUP
        at com.google.android.fhir.datacapture.MoreQuestionnaireItemTypesKt.getValueForType(MoreQuestionnaireItemTypes.kt:41)
        at com.google.android.fhir.datacapture.enablement.EnablementEvaluatorKt.enableWhenTypeToPredicate(EnablementEvaluator.kt:140)
        at com.google.android.fhir.datacapture.enablement.EnablementEvaluatorKt.access$enableWhenTypeToPredicate(EnablementEvaluator.kt:1)
        at com.google.android.fhir.datacapture.enablement.EnablementEvaluator.evaluateEnableWhen(EnablementEvaluator.kt:113)
        at com.google.android.fhir.datacapture.enablement.EnablementEvaluator.evaluate(EnablementEvaluator.kt:70)
        at com.google.android.fhir.datacapture.QuestionnaireViewModel.getQuestionnaireItemViewItemList(QuestionnaireViewModel.kt:115)
        at com.google.android.fhir.datacapture.QuestionnaireViewModel.getQuestionnaireItemViewItemList(QuestionnaireViewModel.kt:127)
        at com.google.android.fhir.datacapture.QuestionnaireViewModel.getQuestionnaireItemViewItemList(QuestionnaireViewModel.kt:127)
        at com.google.android.fhir.datacapture.QuestionnaireViewModel.getQuestionnaireItemViewItemList$datacapture_debug(QuestionnaireViewModel.kt:67)
        at com.google.android.fhir.datacapture.QuestionnaireViewModel$$special$$inlined$map$1$2.emit(Collect.kt:135)
        at kotlinx.coroutines.flow.StateFlowImpl.collect(StateFlow.kt:274)
        at com.google.android.fhir.datacapture.QuestionnaireViewModel$$special$$inlined$map$1.collect(SafeCollector.common.kt:114)
        at com.google.android.fhir.datacapture.QuestionnaireFragment$onViewCreated$1.invokeSuspend(QuestionnaireFragment.kt:73)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at androidx.lifecycle.DispatchQueue.drainQueue(DispatchQueue.kt:75)
        at androidx.lifecycle.DispatchQueue.resume(DispatchQueue.kt:54)
        at androidx.lifecycle.LifecycleController$observer$1.onStateChanged(LifecycleController.kt:40)
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:354)
        at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:265)
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:307)
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:148)
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:134)
        at androidx.fragment.app.FragmentViewLifecycleOwner.handleLifecycleEvent(FragmentViewLifecycleOwner.java:62)
        at androidx.fragment.app.Fragment.restoreViewState(Fragment.java:650)
        at androidx.fragment.app.Fragment.restoreViewState(Fragment.java:2997)
        at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:2988)
        at androidx.fragment.app.FragmentStateManager.activityCreated(FragmentStateManager.java:572)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:281)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2177)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2088)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1990)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3122)
        at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:3056)
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:251)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:473)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1425)
        at android.app.Activity.performStart(Activity.java:7825)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3294)
```

**Comment by pld** [comment]
ah good catch, the `getValueForType` extension doesn't support the group type, here's a simple failing test for this, https://github.com/google/android-fhir/compare/enablement-group-type?expand=1

I'm not sure how `enableWhen` should handle group types, recur into contents?

**Comment by jingtang10** [comment]
Thanks @MuhammadSalman-7214 for creating the issue and adding the details.

@pld I think this is an actual bug. Here https://github.com/google/android-fhir/blob/master/datacapture/src/main/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluator.kt#L71 we shouldn't be passing the type of the question whose enablement status is to be decided. The type there should be the type of the questionnaire response item retrieved using the questionnaireResponseItemRetriver here https://github.com/google/android-fhir/blob/master/datacapture/src/main/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluator.kt#L73.

**Comment by MuhammadSalman-7214** [comment]
your welcome @jingtang10

**Comment by shoaibmushtaq25** [comment]
As I understood, To resolve the issue, we need to pass type of questionaire response item from `questionnaireResponseItemRetriver` here https://github.com/google/android-fhir/blob/master/datacapture/src/main/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluator.kt#L71 instead of passing the type of question `questionnaireItem.type`. Am I understood right? @jingtang10

**Comment by jingtang10** [comment]
@shoaibmushtaq25 yeah that's right. The only problem is that the questionnaire response item doesn't have a type. so maybe one thing you can do is to modify the retriever so it returns a pair of questionnaire item + questionnaire response item.

**Comment by masterishaan19** [comment]
Can I work on this issue ?? Is it still open for work ??

**Comment by shoaibmushtaq25** [comment]
Hey @masterishaan19 , I am working on it and it's about to complete.

**Comment by s-ayush2903** [comment]
Can be closed in #308

**Comment by shoaibmushtaq25** [comment]
PR #308 around this issue has been merged, so closing the issue too.
